zulipterminal/ui_tools/boxes.py

```diff
@@ -619,14 +619,49 @@ def keypress(self, size: urwid_Size, key: str) -> Optional[str]:
                         header.focus_col = self.FOCUS_HEADER_BOX_STREAM
                 else:
                     self.update_recipient_emails(self.to_write_box)
-                    invalid_emails = self.model.get_invalid_recipient_emails(
-                        self.recipient_emails
-                    )
-                    if invalid_emails:
-                        invalid_emails_error = ('Invalid recipient(s) - '
-                                                + ', '.join(invalid_emails))
-                        self.view.set_footer_text(invalid_emails_error, 3)
+                    tidy_recipient_info = ""
+                    invalid_recipients = []
+                    recipients = [recipient.strip()
+                                  for recipient in
+                                  self.to_write_box.edit_text.split(',')
+                                  if recipient.strip()]
+
+                    for recipient in recipients:
+                        cleaned_recipient_list = re.findall(
+                            r"^(.*?)(?:\s*?<?([\w\.-]+@[\w\.-]+)>?(.*))?$",
+                            recipient
+                        )
+                        recipient_name, recipient_email, invalid_text = (
+                            cleaned_recipient_list[0]
+                        )
+                        valid = self.model.check_recipient_validity(
+                            recipient_name, recipient_email
+                        )
+                        tidy_recipient_info += (", " if tidy_recipient_info
+                                                else "")
```

Comment:
This behavior suggests we should use `join` for this too, ie. build a list during this loop.